instance/static/js/src/instance.js

```diff
@@ -122,12 +122,33 @@ app.controller("Index", ['$scope', 'Restangular', 'OpenCraftAPI', '$q',
             } else if(message.data.type === 'instance_log') {
                 if($scope.selected.instance && $scope.selected.instance.id === message.data.instance_id) {
                     $scope.$apply(function(){
-                        $scope.selected.instance.log_text += message.data.log_entry + '\n';
+                        $scope.selected.instance.log_entries.push(message.data.log_entry);
                     });
                 }
             }
         };
 
+        $scope.error_log = function() {
+            if (!$scope.selected.instance) {
+                return;
+            }
+            var log_entries = $scope.selected.instance.log_entries;
+            var error = [];
+            var counting = false;
+            for (var i = log_entries.length; i > 0; i--) {
+                var entry = log_entries[i - 1];
+                if (entry.level != 'ERROR' && entry.level != 'CRITICAL') {
+                    if (counting) {
+                        return error.reverse();
+                    }
```

Comment:
That assumes that the error would always be the last log_entry, which might not always be the case.

I wonder if it wouldn't be cleaner to isolate the error on the server side, when it actually happens, which would be more reliable - a reference to a set of log_entries could be kept when the error is caught from ansible, to ensure there is always a reference to one, and only one, set of error messages when the instance gets into an error state.